StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -582,8 +591,24 @@ extension PaymentSheet {
         return params
     }
 
+    /// - Note: Should only be invoked by merchant code via the `IntentConfiguration.ConfirmHandler.intentCreationCallback`
     private static func _handleIntentCreation(_ result: Result<String, Error>) {
-        // TODO(porter) Deferred intent confirmation handling
+        guard let deferredContext = DeferredIntentContext.current else {
+            fatalError("DeferredIntentContext.current unexpectedly nil")
+        }
+
+        DispatchQueue.main.async {
```

Comment:
Link was crashing if not called on main thread, I think this is better than asking merchant to invoke the callback on the main thread.